paddleseg/models/enet.py

```diff
@@ -0,0 +1,759 @@
+                                                relu=decoder_relu)
+        self.regular5_1 = RegularBottleneck(16,
+                                            padding=1,
+                                            dropout_prob=0.1,
+                                            relu=decoder_relu)
+        self.transposed_conv = nn.Conv2DTranspose(16,
+                                                  num_classes,
+                                                  kernel_size=3,
+                                                  stride=2,
+                                                  padding=1,
+                                                  bias_attr=False)
+
+        self.pretrained = pretrained
+        self.init_weight()
+
+    def forward(self, x):
+        # Initial block
+        # import pdb;pdb.set_trace()
+        # pdb.set_trace()
+        input_size = x.shape
+        x = self.initial_block(x)
+
+        # Stage 1 - Encoder
+        stage1_input_size = x.shape
+        x, max_indices1_0 = self.downsample1_0(x)
+        x = self.regular1_1(x)
+        x = self.regular1_2(x)
+        x = self.regular1_3(x)
+        x = self.regular1_4(x)
+
+        # Stage 2 - Encoder
+        stage2_input_size = x.shape
+        x, max_indices2_0 = self.downsample2_0(x)
+        x = self.regular2_1(x)
+        x = self.dilated2_2(x)
+        x = self.asymmetric2_3(x)
+        x = self.dilated2_4(x)
+        x = self.regular2_5(x)
+        x = self.dilated2_6(x)
+        x = self.asymmetric2_7(x)
+        x = self.dilated2_8(x)
+
+        # Stage 3 - Encoder
+        x = self.regular3_0(x)
+        x = self.dilated3_1(x)
+        x = self.asymmetric3_2(x)
+        x = self.dilated3_3(x)
+        x = self.regular3_4(x)
+        x = self.dilated3_5(x)
+        x = self.asymmetric3_6(x)
+        x = self.dilated3_7(x)
+
+        # Stage 4 - Decoder
+        x = self.upsample4_0(x, max_indices2_0, output_size=stage2_input_size)
+        x = self.regular4_1(x)
+        x = self.regular4_2(x)
+
+        # Stage 5 - Decoder
+        x = self.upsample5_0(x, max_indices1_0, output_size=stage1_input_size)
+        x = self.regular5_1(x)
+        x = self.transposed_conv(x, output_size=input_size[2:])
+        return [x]
+
+    def init_weight(self):
+        if self.pretained is not None:
+            utils.load_pretrained_model(self, self.pretrained)
+
+
+class MaxUnpool2D(nn.Layer):
+    def __init__(self,
+                 kernel_size,
+                 stride=None,
+                 padding=0,
+                 align_corners=False):
+        super(MaxUnpool2D, self).__init__()
+
+        if isinstance(stride, int):
+            self.kernel_size = (kernel_size, kernel_size)
+        else:
+            self.kernel_size = kernel_size
+        if stride is None:
+            self.stride = self.kernel_size
+        else:
+            if isinstance(stride, int):
+                self.stride = (stride, stride)
+            else:
```

Comment:
paddle have implemented maxunpool2d, plz refer to that implementation

Reply:
Done.